### PR TITLE
gh-109190: What's New in 3.12: Add subheadings to removals for easy linking

### PR DESCRIPTION
**Pull request by hugovk**
Add subheadings to the [Removed](https://docs.python.org/3.12/whatsnew/3.12.html#removed) section. 

Similar to [Improved Modules](https://docs.python.org/3.12/whatsnew/3.12.html#improved-modules), this gives us hyperlink anchors for the subheadings and makes it easy to share direct links. For example, I wanted to share a direct link to the `ssl.match_hostname()` removal, but had to use https://docs.python.org/3.12/whatsnew/3.12.html#removed which is a bit further away.

Before: https://docs.python.org/3.12/whatsnew/3.12.html#removed

After: https://cpython-previews--109159.org.readthedocs.build/en/109159/whatsnew/3.12.html#removed

----
:books: Documentation preview :books:: https://cpython-previews--109159.org.readthedocs.build/en/109159/whatsnew/3.12.html#removed

* Issue: gh-109190

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -1274,29 +1274,41 @@ although there is currently no date scheduled for their removal.
...
+* :mod:`configparser` no longer has a ``SafeConfigParser`` class. Use the
+  shorter :class:`~configparser.ConfigParser` name instead.
+* :class:`configparser.ConfigParser` no longer has a ``readfp`` method.
+  Use :meth:`~configparser.ConfigParser.read_file` instead.
+
+distutils
+---------
 
-* ``distutils``: Remove the ``distutils`` package. It was deprecated in Python 3.10 by
+* Remove the ``distutils`` package. It was deprecated in Python 3.10 by
```
Comment (AA-Turner) [inline Doc/whatsnew/3.12.rst]:
```suggestion
* Remove the :py:mod:`!distutils` package. It was deprecated in Python 3.10 by
```

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -1274,29 +1274,41 @@ although there is currently no date scheduled for their removal.
...
   Use :mod:`asyncio` instead.
   (Contributed by Nikita Sobolev in :gh:`96580`.)
 
-* :mod:`configparser`: Several names deprecated in the :mod:`configparser` way back in 3.2 have
-  been removed per :gh:`89336`:
+configparser
+------------
+
+Several names deprecated in the :mod:`configparser` way back in 3.2 have
+been removed per :gh:`89336`:
```
Comment (AA-Turner) [inline Doc/whatsnew/3.12.rst]: Should we use bullets for these new sections? E.g. async* above does.

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -1314,46 +1326,64 @@ Removed
...
+enum
+----
+
+* Remove :mod:`enum`'s ``EnumMeta.__getattr__``, which is no longer needed for
   enum attribute access.
   (Contributed by Ethan Furman in :gh:`95083`.)
 
-* :mod:`ftplib`: Remove the ``FTP_TLS.ssl_version`` class attribute: use the
+ftplib
+------
```
Comment (AA-Turner) [inline Doc/whatsnew/3.12.rst:1337]: By moving bullets to subheads we loose the module links -- should we add them back? (Happy to tackle this in a follow-up PR if this is agreed)

**Review by AA-Turner — APPROVED** [review]
Good idea!

A

File: Doc/whatsnew/3.12.rst (hunk 2 above)
Comment (hugovk) [reply Doc/whatsnew/3.12.rst]: I kept the original bullets for most of these sections, that only had a single level of bullets.

A few (at least this, importlib and unittest) were of the form:

---

* configparser: Several names deprecated ... have been removed:

  * first one
  * second
  * third
---

And I changed them to:

---

## configparser

Several names deprecated ... have been removed:

* first one
* second
* third

---

But on second thoughts, it's probably better to keep the original two-level bullets for consistency, so things are all nicely aligned as you scroll down?

File: Doc/whatsnew/3.12.rst (hunk 3 above)
Comment (hugovk) [reply Doc/whatsnew/3.12.rst:1337]: The module link is useful but I think we should avoid repeating them:

---

## ftplib

* ftplib: Remove the ``FTP_TLS.ssl_version`` class attribute: use the *context* parameter instead.

---

I had generally slurped the module link into the main text (e.g. enum), when there wasn't one already there (e.g. importlib).

Looks like I missed one or two, let me check and add.

Comment (hugovk) [reply Doc/whatsnew/3.12.rst:1337]: Updated, how's that?

File: Doc/whatsnew/3.12.rst (hunk 2 above)
Comment (AA-Turner) [reply Doc/whatsnew/3.12.rst]: I'd agree--no strong opinions on keeping / removing, it was just the consistency point.

A

File: Doc/whatsnew/3.12.rst (hunk 3 above)
Comment (AA-Turner) [reply Doc/whatsnew/3.12.rst:1337]: Fab, thanks

File: Doc/whatsnew/3.12.rst (hunk 2 above)
Comment (hugovk) [reply Doc/whatsnew/3.12.rst]: Thanks, reindented.

File: Doc/whatsnew/3.12.rst [hunk 4]
```diff
@@ -1361,21 +1361,21 @@ hashlib
 importlib
 ---------
 
-Many previously deprecated cleanups in :mod:`importlib` have now been
-completed:
+  Many previously deprecated cleanups in :mod:`importlib` have now been
```
Comment (AA-Turner) [inline Doc/whatsnew/3.12.rst]: Pure indentation will render as a block-quote: https://cpython-previews--109159.org.readthedocs.build/en/109159/whatsnew/3.12.html#importlib
```suggestion
* Many previously deprecated cleanups in :mod:`importlib` have now been
```

**Review by AA-Turner — APPROVED** [review]

**Pull request merged**